src/ucp/api/ucp.h

```diff
@@ -629,6 +631,54 @@ void ucp_worker_release_address(ucp_worker_h worker, ucp_address_t *address);
  */
 void ucp_worker_progress(ucp_worker_h worker);
 
+/**
+ * @ingroup UCP_WORKER
+ * @brief Obtain an event file descriptor for interrupt notification.
+ *
+ * This routine returns a valid file descriptor for polling functions.
+ * The file descriptor will get signaled on any event on one of the underlying
+ * Transports.
+ *
+ * @param [in]  worker    Worker of notified events.
+ * @param [out] fd        File descriptor.
+ */
+ucs_status_t ucp_worker_get_efd(ucp_worker_h worker, int *fd_p);
+
+
+/**
+ * @ingroup UCP_WORKER
+ * @brief Turn on interrupt notification for the next event.
+ *
+ * This routine needs to be called for every notification on this worker, so
+ * will typically be called once the processing of the revious event is over.
+ * It must also be called after @ref ucp_worker_event_get_fd for activation.
+ *
+ * @param [in]  worker    Worker of notified events.
+ */
+ucs_status_t ucp_worker_arm(ucp_worker_h worker);
+
+
+/**
+ * @ingroup UCP_WORKER
+ * @brief Wait for an interrupt notification.
```

Comment:
needs rephrasing, interrupt is an implementation detail, need to describe in terms of API